src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -700,7 +714,9 @@ private void validateShare(Path sharePath) throws IOException {
     // Verify that the adaptor has permission to read the Acl and share Acl.
     try {
       readShareAcls(sharePath);
-      delegate.getAclViews(sharePath);
+      if (!delegate.isDfsNamespace(sharePath) || allowFilesInDfsNamespaces) {
+        delegate.getAclViews(sharePath);
```

Comment:
Should allowFilesInDfsNamespaces be per start path?   If allowFilesInDfsNamespaces does not have to be per start path then why not always do getAclViews for every dfs-namespace and non-dfs-namespace?